### PR TITLE
SAK-40921 - Fix content Item / deep link flow

### PR DESCRIPTION
**Pull request by csev**

### CODE REVIEW

File: basiclti/basiclti-blis/src/java/org/sakaiproject/lti13/LTI13Servlet.java [hunk 1]
```diff
@@ -1548,24 +1542,30 @@ private void handleLineItemsDelete(String signed_placement, String lineItem, Htt
 	private void handleOIDCAuthorization(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
 		String state = (String) request.getParameter("state");
-		if ( state != null && state.trim().length() < 1 ) state = null;
+		if ( StringUtils.isEmpty(state) ) state = null;
```
Comment (ottenhoff) [inline basiclti/basiclti-blis/src/java/org/sakaiproject/lti13/LTI13Servlet.java]: state = StringUtils.trimToNull(state)

File: basiclti/basiclti-impl/src/java/org/sakaiproject/basiclti/impl/BasicLTISecurityServiceImpl.java [hunk 2]
```diff
@@ -262,15 +262,22 @@ private void doSplash(HttpServletRequest req, HttpServletResponse res, String sp
...
-	private void redirectOIDC(HttpServletRequest req, HttpServletResponse res, String oidc_endpoint, ResourceLoader rb)
+	private void redirectOIDC(HttpServletRequest req, HttpServletResponse res, Map<String, Object> tool, String oidc_endpoint, ResourceLoader rb)
 	{
 		// req.getRequestURL()=http://localhost:8080/access/basiclti/site/85fd092b-1755-4aa9-8abc-e6549527dce0/content:0
 		// req.getRequestURI()=/access/basiclti/site/85fd092b-1755-4aa9-8abc-e6549527dce0/content:0
 		String login_hint = req.getRequestURI().toString();
-		// TODO: Encrypt the login hint
+		String query_string = req.getQueryString();
+		if ( StringUtils.isNotEmpty(query_string)) {
+			login_hint = login_hint + "?" + query_string;
```
Comment (ottenhoff) [inline basiclti/basiclti-impl/src/java/org/sakaiproject/basiclti/impl/BasicLTISecurityServiceImpl.java]: Seems more clear to show your concatenation:

login_hint += "?" + query_string;

**Comment by mpellicer** [comment]
@csev WIP?

File: basiclti/basiclti-blis/src/java/org/sakaiproject/lti13/LTI13Servlet.java (hunk 1 above)
Comment (csev) [reply basiclti/basiclti-blis/src/java/org/sakaiproject/lti13/LTI13Servlet.java]: Fixed

**Pull request merged**